### PR TITLE
Update UNITTESTS and Greentea test docs

### PR DESCRIPTION
**Pull request by rajkan01**
- Updated UNITTESTS and greentea test docs according to a new directory structure

### CODE REVIEW

File: docs/debugging-testing/testing/testing_greentea.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ This document will help you start using Greentea. Please see the [`htrun` docume
 
 ### Test code structure
 
-You can run tests throughout Mbed OS and for your project's code. They are located under a special directory called `TESTS`.
+You can run tests throughout Mbed OS and for your project's code. They are located under a special directory called `TESTS`of each library.
```
Comment (evedon) [inline docs/debugging-testing/testing/testing_greentea.md]:
```suggestion
Greentea tests can be created anywhere in the Mbed OS directory structure or in your project's code. They are located under a special directory called `TESTS`. By convention, greentea tests for a specific Mbed OS library should be placed in a `TESTS` folder under that library. The `mbed-os/TESTS` folder is dedicated to integration or system tests.
```

File: docs/debugging-testing/testing/testing_greentea.md [hunk 2]
```diff
@@ -27,10 +27,10 @@ int main() {
 In addition to being placed under a `TESTS` directory, test sources must exist under two other directories: a test group directory and a test case directory. The following is an example of this structure:
 
 ```
-myproject/TESTS/test_group/test_case_1
+mbed-os/mylibrary/tests/TESTS/test_group/test_case_1
```
Comment (evedon) [inline docs/debugging-testing/testing/testing_greentea.md]: I would not change the example with `myproject`; it is still valid and it was clearer.

File: docs/debugging-testing/testing/testing_greentea.md [hunk 3]
```diff
@@ -27,10 +27,10 @@ int main() {
...
 ```
 
-In this example, `myproject` is the project root, and all the source files under the `test_case_1` directory are included in the test. The test build also includes any other source files from the OS, libraries and projects that apply to the target's configuration.
+In this example, `mylibrary` is like platform, drivers etc., and all the source files under the `test_case_1` directory are included in the test. The test build also includes any other source files from the OS, libraries and projects that apply to the target's configuration.
```
Comment (evedon) [inline docs/debugging-testing/testing/testing_greentea.md]: Similarly, keep the original text.

File: docs/debugging-testing/testing/testing_greentea.md [hunk 4]
```diff
@@ -54,7 +54,7 @@ Tests can also be completely ignored by using the `.mbedignore` file described i
 
 #### Test names
 
-A test case is named by its position in your project's file structure. For instance, in the above example, a test case with the path `myproject/TESTS/test_group/test_case_1` would be named `tests-test_group-test_case_1`. The name is created by joining the directories that make up the path to the test case with a dash `-` character. This is a unique name to identify the test case. You will see this name throughout the build and testing process.
+A test case is named by its position in your project's file structure. For instance, in the above example, a test case with the path `mylibrary/tests/TESTS/test_group/test_case_1` would be named `tests-test_group-test_case_1`. The name is created by joining the directories that make up the path to the test case with a dash `-` character. This is a unique name to identify the test case. You will see this name throughout the build and testing process.
```
Comment (evedon) [inline docs/debugging-testing/testing/testing_greentea.md]: Keep original.

File: docs/debugging-testing/testing/testing_greentea.md [hunk 5]
```diff
@@ -54,7 +54,7 @@ Tests can also be completely ignored by using the `.mbedignore` file described i
...
 
 ### Building tests
```
Comment (evedon) [inline docs/debugging-testing/testing/testing_greentea.md]: Adding comment here for the rest of the file:
- Section Writing integration tests using host tests: update the location of host tests to `mbed-os/TESTS/host_tests` and sy that this directory should be created if it does not exist.
- Output extracts should be updated with new directory structure

**Review by evedon — CHANGES_REQUESTED** [review]

**Comment by evedon** [comment]
You missed some instances of `tests-mbedmicro-rtos-mbed-mail`. I'd suggest to run the actual commands and copy & paste the output.

**Comment by Elise-Kaminski** [comment]
LGTM. I've just made some minor edits.

**Review by evedon — APPROVED** [review]

**Comment by Elise-Kaminski** [comment]
Looks good. I reapplied the changes I had done.

**Pull request merged**